Update hardware-and-software-requirements.md (#19360)
pingcap/docs: hardware-and-software-requirements.md

> - In the production environment, the TiDB and PD instances can be deployed on the same server. If you have a higher requirement for performance and reliability, try to deploy them separately.
> - It is strongly recommended to configure TiDB, TiKV, and TiFlash with at least 8 CPU cores each in the production environment. To get better performance, a higher configuration is recommended.
> - It is recommended to keep the size of TiKV hard disk within 4 TB if you are using PCIe SSDs or within 1.5 TB if you are using regular SSDs.

> > - If you deploy your TiKV on a cloud provider, such as AWS, Google Cloud, or Azure, TiKV nodes must use cloud disks. Data on local disks will be lost if the instance crashes in the cloud environment.

Author: ```suggestion
> - If you deploy your TiKV on a cloud provider, such as AWS, Google Cloud, or Azure, It is recommended to use cloud disks for TiKV nodes. Data on local disks will be lost if the instance crashes in the cloud environment.
```

> > - If you deploy your TiDB cluster on a cloud provider, such as AWS, Google Cloud, or Azure, TiKV nodes must use cloud disks. Data on local disks will be lost if the instance crashes in the cloud environment.

Author: ```suggestion
> - If you deploy your TiKV on a cloud provider, such as AWS, Google Cloud, or Azure, TiKV nodes must use cloud disks. Data on local disks will be lost if the instance crashes in the cloud environment.
```

> > - If you deploy TiKV on a cloud provider, such as AWS, Google Cloud, or Azure, it is recommended to use cloud disks for TiKV nodes. Data on local disks might be lost if the instance crashes in the cloud environment.

Reviewer: ```suggestion
> - If you deploy TiKV on a cloud provider, such as AWS, Google Cloud, or Azure, it is recommended to use cloud disks for TiKV nodes. Data on local disks might be lost if the TiKV instance crashes in the cloud environment.
```

> > - If you deploy your TiKV on a cloud provider, such as AWS, Google Cloud, or Azure, it is recommended to use cloud disks for TiKV nodes. Data on local disks will be lost if the instance crashes in the cloud environment.

Author: ```suggestion
> - If you deploy your TiKV on a cloud provider, such as AWS, Google Cloud, or Azure, it is recommended to use cloud disks for TiKV nodes. Data on local disks might be lost if the instance crashes in the cloud environment.
```

> > - If you deploy your TiKV on a cloud provider, such as AWS, Google Cloud, or Azure, it is recommended to use cloud disks for TiKV nodes. Data on local disks might be lost if the instance crashes in the cloud environment.

Author: ```suggestion
> - If you deploy TiKV on a cloud provider, such as AWS, Google Cloud, or Azure, it is recommended to use cloud disks for TiKV nodes. Data on local disks might be lost if the instance crashes in the cloud environment.
```
